docs/snippets/vue/component-story-custom-args-icons.vue3.js.mdx

```diff
@@ -0,0 +1,39 @@
+```js
+// YourComponent.stories.js
+
+import YourComponent from './YourComponent.vue';
+
+import { IconA, IconB, IconC, IconD, IconE } from './icons';
+
+//👇 Maps the icons to a JSON serializable object to be safely used with the argTypes
+const iconMap = { IconA, IconB, IconC, IconD, IconE };
+
+export default {
+  title: 'My Story with Icons',
+  component: YourComponent,
+  //👇 Creates specific argTypes with options
+  argTypes: {
+    icon: {
+      control: {
+        type: 'select',
+        options: Object.keys(iconMap),
+      },
+    },
+  },
+};
+
+const Template = (args) => {
+  return {  
+    components: { YourComponent },
+    setup() {
+      //👇 The args will now be passed down to the component
+      return {
+        ...args,
+        //👇 Replaces the local variable with the override (without mutating it)
+        icon: iconMap[args.icon],
+      };
```

Comment:
```suggestion
      return {
        args: {
          ...args,
          //👇 Replaces the local variable with the override (without mutating it)
          icon: iconMap[args.icon],
        }
      };
```

Same as above

Reply:
@phated fixed with the latest commit. Thanks for the clarification.